src/HealthChecks.AzureServiceBus/AzureEventHubHealthCheck.cs

```diff
@@ -12,34 +12,48 @@ public class AzureEventHubHealthCheck
     {
         private static readonly ConcurrentDictionary<string, EventHubClient> _eventHubConnections = new ConcurrentDictionary<string, EventHubClient>();
 
-        private readonly string _eventHubConnectionString;
+        public string EntityPath => ConnectionStringBuilder.EntityPath;
+        public Uri Endpoint => ConnectionStringBuilder.Endpoint;
+        protected EventHubsConnectionStringBuilder ConnectionStringBuilder { get; }
+
         public AzureEventHubHealthCheck(string connectionString, string eventHubName)
         {
             if (string.IsNullOrEmpty(connectionString))
             {
                 throw new ArgumentNullException(nameof(connectionString));
-            }
-
+            }
+
             if (string.IsNullOrEmpty(eventHubName))
             {
                 throw new ArgumentNullException(nameof(eventHubName));
             }
 
-            var builder = new EventHubsConnectionStringBuilder(connectionString)
```

Comment:
I don't understand the difference between old code and new one, do the same thing with a new overload to specify the connection string without the EntityPath. ¿What issue is trying to solve?

Reply:
See https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/issues/469

We're providing capability to use event hub name as part of the connection string by using `EventHubsConnectionStringBuilder` and using that to authenticate with.